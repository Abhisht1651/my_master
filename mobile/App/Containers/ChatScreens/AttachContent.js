// @flow
import React from 'react'
import { View, Text, TouchableOpacity, Modal, Image } from 'react-native'
import Images from '../../Themes/Images'
import { MediumThinnerButton } from '../../Components/Buttons/MediumThinnerButton'

// Styles
import styles from './ScreenStyles/AttachContentStyles'

type Props = {
  isModalVisible: boolean,
  onClose: () => void
}

const AttachContent = (props: Props) => {
  const { isModalVisible, onClose } = props
  return (
    <Modal visible={isModalVisible} transparent>
      <View style={styles.screen}>
        <View style={styles.innerScreen}>
          <View style={styles.view1}>
            <Text style={styles.headerText}>Attach Content</Text>
            <TouchableOpacity style={styles.closeIconContainer} activeOpacity={1} onPress={onClose}>
              <Image source={Images.closeIcon} style={styles.closeIcon} />
            </TouchableOpacity>
          </View>
          <View style={styles.view2}>
            <View style={styles.space2} />
            <MediumThinnerButton title="Storyline" outline />
            <View style={styles.space} />
            <MediumThinnerButton title="Entity" outline />
            <View style={styles.space} />
            <MediumThinnerButton title="Poll" outline />
            <View style={styles.space} />
            <MediumThinnerButton title="Media" outline />
          </View>
        </View>
        <View style={styles.innerScreen2} />
      </View>
    </Modal>
  )
}
AttachContent.defaultProps = {
  isModalVisible: true
}

export default AttachContent
