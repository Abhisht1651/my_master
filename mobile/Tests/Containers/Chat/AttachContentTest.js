import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image } from 'react-native'
import AttachContent from '../../../App/Containers/ChatScreens/AttachContent'

describe('AttachContent', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<AttachContent />) // no compile-time error
  })
  it('AttachContent component renders correctly', () => {
    const wrapper = shallow(<AttachContent />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(1)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
})
