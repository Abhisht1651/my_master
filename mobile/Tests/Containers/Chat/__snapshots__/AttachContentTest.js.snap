// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachContent AttachContent component renders correctly 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#2222234D",
        "flexDirection": "column-reverse",
        "height": 1334,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 533.9285714285714,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E6ECED",
            "flexDirection": "row",
            "height": 105.14285714285714,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Medium",
              "fontSize": 18,
              "letterSpacing": -0.12,
              "marginLeft": "auto",
              "textAlign": "right",
              "width": "50%",
            }
          }
        >
          Attach Content
        </Text>
        <ForwardRef
          activeOpacity={1}
          style={
            Object {
              "marginLeft": 95.75,
              "marginRight": 25.97,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "aspectRatio": 1,
                "height": 17.28,
              }
            }
          />
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 19.714285714285715,
            }
          }
        />
        <MediumThinnerButton
          onPress={[Function]}
          outline={true}
          style={Object {}}
          title="Storyline"
        />
        <View
          style={
            Object {
              "margin": 9.857142857142858,
            }
          }
        />
        <MediumThinnerButton
          onPress={[Function]}
          outline={true}
          style={Object {}}
          title="Entity"
        />
        <View
          style={
            Object {
              "margin": 9.857142857142858,
            }
          }
        />
        <MediumThinnerButton
          onPress={[Function]}
          outline={true}
          style={Object {}}
          title="Poll"
        />
        <View
          style={
            Object {
              "margin": 9.857142857142858,
            }
          }
        />
        <MediumThinnerButton
          onPress={[Function]}
          outline={true}
          style={Object {}}
          title="Media"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2222234D",
          "height": 800.0714285714286,
        }
      }
    />
  </View>
</Component>
`;
